File: backend/app/schemas/knowledge_config.py
"""Pydantic schemas for KnowledgeConfig CRUD."""

from typing import Optional

from pydantic import BaseModel, Field


class KnowledgeConfigCreate(BaseModel):
    """Schema for creating a KnowledgeBase configuration."""

    knowledge_base_id: int = Field(..., description="KnowledgeBase ID")
    chunk_size: Optional[int] = Field(
        default=None, ge=50, le=5000, description="Chunk size in characters"
    )
    chunk_overlap: Optional[int] = Field(
        default=None, ge=0, le=1000, description="Chunk overlap in characters"
    )
    embedding_provider: Optional[str] = Field(
        default=None, max_length=50, description="Embedding provider type"
    )
    embedding_model: Optional[str] = Field(
        default=None, max_length=200, description="Embedding model identifier"
    )
    retrieval_top_k: Optional[int] = Field(
        default=None, ge=1, le=100, description="Top‑K for retrieval"
    )


class KnowledgeConfigUpdate(BaseModel):
    """Schema for updating a KnowledgeBase configuration. All fields optional."""

    chunk_size: Optional[int] = Field(
        default=None, ge=50, le=5000, description="Chunk size in characters"
    )
    chunk_overlap: Optional[int] = Field(
        default=None, ge=0, le=1000, description="Chunk overlap in characters"
    )
    embedding_provider: Optional[str] = Field(
        default=None, max_length=50, description="Embedding provider type"
    )
    embedding_model: Optional[str] = Field(
        default=None, max_length=200, description="Embedding model identifier"
    )
    retrieval_top_k: Optional[int] = Field(
        default=None, ge=1, le=100, description="Top‑K for retrieval"
    )


class KnowledgeConfigResponse(BaseModel):
    """Schema for KnowledgeConfig API response."""

    id: int
    knowledge_base_id: int
    chunk_size: Optional[int] = None
    chunk_overlap: Optional[int] = None
    embedding_provider: Optional[str] = None
    embedding_model: Optional[str] = None
    retrieval_top_k: Optional[int] = None
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True

File: frontend/src/api/knowledgeConfig.ts
import apiClient from './client'

export interface KnowledgeConfig {
  knowledge_base_id: number
  chunk_size: number
  chunk_overlap: number
  embedding_provider: string | null
  embedding_model: string
  retrieval_top_k: number
  id: number | null
  created_at: string | null
  updated_at: string | null
}

export interface KnowledgeConfigUpdate {
  chunk_size?: number
  chunk_overlap?: number
  embedding_provider?: string | null
  embedding_model?: string
  retrieval_top_k?: number
}

export async function getKnowledgeConfig(knowledgeBaseId: number): Promise<KnowledgeConfig> {
  const { data } = await apiClient.get(`/knowledge-bases/${knowledgeBaseId}/config`)
  return data
}

export async function updateKnowledgeConfig(
  knowledgeBaseId: number,
  payload: KnowledgeConfigUpdate,
): Promise<KnowledgeConfig> {
  const { data } = await apiClient.put(`/knowledge-bases/${knowledgeBaseId}/config`, payload)
  return data
}